internal/internal_task_handlers.go

```diff
@@ -1670,7 +1670,7 @@ func (i *cadenceInvoker) Heartbeat(details []byte) error {
 			i.Unlock()
 
 			if detailsToReport != nil {
-				i.Heartbeat(*detailsToReport)
+				i.Heartbeat(*detailsToReport, false)
```

Comment:
As you mentioned offline, put a TODO here to revisit the race condition issue

Reply:
Done.